File: tests/C/0012.c
int main (int argc, char *argv[])
{
	#pragma acc kernels
	{
		int oneliner_stmt = 0;
		#pragma acc loop
		for (int i = 0; i < 10; ++i)
		{
			#pragma acc loop
			for (int j = 0; j < 10; ++j)
			{
			}
		}
		#pragma acc loop
		for (int i = 0; i < 10; ++i)
			#pragma acc loop
			for (int j = 0; j < 10; ++j)
				oneliner_stmt++;
		oneliner_stmt++;
		oneliner_stmt++;
		oneliner_stmt++;
		#pragma acc loop
		for (int i = 0; i < 10; ++i)
			#pragma acc loop
			for (int j = 0; j < 10; ++j)
				#pragma acc loop
				for (int k = 0; k < 3; ++k)
					oneliner_stmt++;
		#pragma acc loop
		for (int i = 0; i < 10; ++i) {
			#pragma acc loop
			for (int j = 0; j < 10; ++j)
				oneliner_stmt++;
		}
		#pragma acc loop
		for (int i = 0; i < 10; ++i)
			#pragma acc loop
			for (int j = 0; j < 10; ++j)
			{
			}
		#pragma acc loop
		for (int i = 0; i < 10; ++i)
		{
			for (int j = 0; j < 10; ++j)
			{
				#pragma acc loop
				for (int k = 0; k < 3; ++k)
					oneliner_stmt++;
			}
			#pragma acc loop
			for (int j = 0; j < 10; ++j)
			{
			}
		}
	}
}

File: tests/Fortran/Free/0012.f90
program test
implicit none
integer i, j, k, l, m

  !$acc kernels
  !$acc loop
  do i = 1, 10
    !$acc loop
    do j = 1, 10
    end do
  end do
  l = 1
  m = 2
  !$acc loop
  do i = 1, 10
    !$acc loop
    do j = 1, 10
    end do
  end do
  !$acc loop
  do i = 1, 10
    !$acc loop
    do j = 1, 10
      do k = 1, 3
      enddo
    enddo
    !$acc loop
    do j = 1, 10
    enddo
  enddo
  !$acc end kernels

end program
